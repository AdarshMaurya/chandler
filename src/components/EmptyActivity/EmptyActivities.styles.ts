import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: 'white',
  },
  image: {
    marginBottom: 20,
  },
  textContainer: {
    paddingHorizontal: 60,
  },
  title: {
    textAlign: 'center',
    marginBottom: 10,
  },
  subtitle: {
    fontSize: 13,
    textAlign: 'center',
  },
});
