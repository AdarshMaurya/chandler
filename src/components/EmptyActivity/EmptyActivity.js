import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {View, Text, Image} from 'react-native';

import {commonStyles} from 'theme';
import {styles} from './EmptyActivities.styles';

export class EmptyActivity extends PureComponent {
  static propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    image: PropTypes.number.isRequired,
  };
  render() {
    const {title, subtitle, image} = this.props;
    return (
      <View style={styles.container}>
        <View style={commonStyles.flex} />

        <Image source={image} style={styles.image} />
        <View style={styles.textContainer}>
          <Text style={styles.title}>{title}</Text>
          {subtitle && <Text style={styles.subtitle}>{subtitle}</Text>}
        </View>

        <View style={commonStyles.flex} />
      </View>
    );
  }
}
