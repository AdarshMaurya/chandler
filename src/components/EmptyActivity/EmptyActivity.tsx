import React, {PureComponent} from 'react';
import {Image, Text, View} from 'react-native';

import {commonStyles} from '@theme';
import {styles} from './EmptyActivities.styles';

interface IEmptyActivityProps {
  title: string;
  subtitle?: string;
  image: number | {uri: string};
}

export class EmptyActivity extends PureComponent<IEmptyActivityProps> {
  public render() {
    const {title, subtitle, image} = this.props;
    return (
      <View style={styles.container}>
        <View style={commonStyles.flex} />

        <Image source={image} style={styles.image} />
        <View style={styles.textContainer}>
          <Text style={styles.title}>{title}</Text>
          {subtitle && <Text style={styles.subtitle}>{subtitle}</Text>}
        </View>

        <View style={commonStyles.flex} />
      </View>
    );
  }
}
