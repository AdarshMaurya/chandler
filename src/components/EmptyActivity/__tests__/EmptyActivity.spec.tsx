import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {EmptyActivity} from '../EmptyActivity';

describe('Components', () => {
  describe('EmptyActivity', () => {
    it('should renders correctly with no subtitle', () => {
      expect(
        toJson(shallow(<EmptyActivity image={2} title="My title" />))
      ).toMatchSnapshot();
    });

    it('should renders correctly with a subtitle', () => {
      expect(
        toJson(
          shallow(
            <EmptyActivity image={2} title="My title" subtitle="My subtitle" />
          )
        )
      ).toMatchSnapshot();
    });
  });
});
