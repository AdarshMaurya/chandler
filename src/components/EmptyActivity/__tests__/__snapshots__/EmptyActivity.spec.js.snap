// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components EmptyActivity should renders correctly with a subtitle 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <Image
    source={2}
    style={
      Object {
        "marginBottom": 20,
      }
    }
  />
  <View
    style={
      Object {
        "paddingHorizontal": 60,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      My title
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 13,
          "textAlign": "center",
        }
      }
    >
      My subtitle
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`Components EmptyActivity should renders correctly with no subtitle 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <Image
    source={2}
    style={
      Object {
        "marginBottom": 20,
      }
    }
  />
  <View
    style={
      Object {
        "paddingHorizontal": 60,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      My title
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
